tests/trainer/connectors/test_callback_connector.py

```diff
@@ -89,10 +87,8 @@ def test_attach_model_callbacks():
     """Test that the callbacks defined in the model and through Trainer get merged correctly."""
 
     def assert_composition(trainer_callbacks, model_callbacks, expected):
-        model = Mock()
-        model.configure_callbacks.return_value = model_callbacks
         trainer = Trainer(checkpoint_callback=False, progress_bar_refresh_rate=0, callbacks=trainer_callbacks)
-        trainer.model = model
+        trainer.call_hook = Mock(return_value=model_callbacks)
```

Comment:
why can we no longer mock the method on the module?

Reply:
Because the previous implementation of `_attach_model_callbacks` did:

`self.trainer.model.configure_callbacks()`

but now it is

`self.trainer.call_hook("configure_callbacks")`


which means we can just mock the call hook function

Reply:
Or are you asking why this changed? It was failing in the original implementation of this PR but seems like it's not anymore.

Do you prefer I undo these changes?

Reply:
is call_hook incompatible with mock?

if not, then mocking the call_hook here makes not much sense as this would result in it returning the model callbacks always, for all hook calls.

Reply:
`call_hook` is compatible with mock.

> this would result in it returning the model callbacks always, for all hook calls.

True, but note that these are unit tests and just test the return value of a method in the connector.

Reply:
Reverted these changes to keep the original pattern of the test. However, we cannot use a mock for the model anymore because of the `is_overridden` check in `call_hook`